### PR TITLE
Water body auto clip-include

### PR DESCRIPTION
**Pull request by huwb**
Make waterbody automatically clip-include

If clipping is enabled and clipping set to remove everything by default, then the waterbody ensures its area is clip-included.

This seems good to me, but needs the ocean lifetime stuff to ensure OnEnable is called after the OceanRenderer.

No urgency on this so ill just park it here.

NOTE: Just realised i created it against master instead of local-water-body. I'm guessing the life cycle stuff won't merge soon so this is probably fine, but review-by-commit is necessary, just the last one is relevant.

### CODE REVIEW

**Comment by huwb** [comment]
You mean you are setting to clip everything by default, then adding a box to include, then a sphere to cut out again?

File: crest/Assets/Crest/Crest/Scripts/WaterBody.cs
Comment (huwb) [reply crest/Assets/Crest/Crest/Scripts/WaterBody.cs]: thanks fixed

File: crest/Assets/Crest/Crest/Scripts/WaterBody.cs
Comment (huwb) [reply crest/Assets/Crest/Crest/Scripts/WaterBody.cs:142]: done now

Comment (daleeidd) [reply crest/Assets/Crest/Crest/Scripts/WaterBody.cs:142]: Thanks

**Review by daleeidd — APPROVED** [review]
Nice work.

**Review by daleeidd — CHANGES_REQUESTED** [review]
I missed this but the execution order will need to be after the OR for it to work in builds. Can merge after that is done.

**Review by daleeidd — APPROVED** [review]
I did the last review items. Ready to merge when you're ready.

**Comment by daleeidd** [comment]
@huwb happy to resolve release note conflict and merge if you're fine with everything.

**Comment by huwb** [comment]
thanks for bumping

what is the requirement for the update order? i'm not super against it but just curious if we can avoid it. the update order adds implicit dependencies between parts of the code which are not documented and can break, so previously i did a push to minimise the use of update orders.

perhaps an alternative is to call our own update function on all waterbodies from the OceanRenderer component. however im aware this is editor code and maybe not worth it. but it might be good to record what the update order dependence is in a comment.

**Comment by daleeidd** [comment]
> what is the requirement for the update order?

The Update method is stripped from builds. So this feature will only get triggered from OnEnable in builds which is execution order dependent on OR.

> but it might be good to record what the update order dependence is in a comment.

Done.

> perhaps an alternative is to call our own update function on all waterbodies from the OceanRenderer component.

Agreed. Going to look at doing that next. I am considering using C# delegates or would you prefer we maintain our own registry and call directly?

**Comment by huwb** [comment]
Ohh it's OnEnable. I guess this is part of the same update issue we're discussing oh the other pr that ensures events get called at the right times when OR enabled/disabled. Do you know which PR I mean? I'll see if I can find it after posting this.

**Comment by huwb** [comment]
Ok yeah #629 . ideally we could converge that rather than workaround it here. I'll try to reply there shortly.

I'd be fine with the changes as is in the meantime - using update order - but would be good to revert once #629 merges. or we could try to get #629 done first.

**Pull request merged**
